### ISSUE TITLE
Add option to keep data in PSD Data Structure

### ISSUE THREAD

**Issue by jdpigeon** [neurosity/eeg-pipes#26]
currently, when going from Epoch to PSD we leave behind the data array that contains the raw EEG data. Some people may want to keep this around in their processing pipeline, but for sake of efficiency, maybe we should just make this an option?

**Comment by ruut12** [comment]
yes please :)
Any workaround atm?

**Comment by kylemath** [comment]
yes same problem we are just hitting, want to plot raw data and spectra side by side for example, how do we pull out data from the intermediate pipe steps

**Comment by beebus** [comment]
Would the tap() operator (formerly the do() operator) work for this?

I just finished a tutorial on RxJS and I used the following code which pulled data from an intermediate pipe step, using the tap() operator. Now, it just console.logged it, but I assume that you could do other things with that data using the tap() operator.

> this.results$ = this.searchSubject$
      .pipe(debounceTime(200))
      .pipe(distinctUntilChanged())
      .pipe(tap(x => console.log('tap', x)))
      .pipe(switchMap(searchString => this.queryAPI(searchString)));

You can see the results of this in the console, each time it logs "tap ... ".
![Picture1](https://user-images.githubusercontent.com/3438663/97055347-41c41780-154c-11eb-902f-3d817ff7b84d.png)

**Comment by beebus** [comment]
And if the question is "How do we **go back to** that data that was in those intermediate steps?", then I would think either logging the data to a database that can be retrieved later or (if possible) some sort of repeating of those same pipe steps to get the same data again.
